PerimeterXModule/Internals/CredentialsIntelligence/ExtractorObject.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Runtime.Serialization;
+
+namespace PerimeterX
+{
+   [DataContract]
+   public class ExtractorObject
+   {
+        [DataMember(Name = "path")]
+        public string Path;
+
+        [DataMember(Name = "path_type")]
+        public string PathType;
+
+        [DataMember(Name = "method")]
+        public string Method;
+
+        [DataMember(Name = "sent_through")]
+        public string SentThrough;
+
+        [DataMember(Name = "pass_field")]
+        public string PassFieldName;
+
+        [DataMember(Name = "user_field")]
+        public string UserFieldName;
```

Comment:
Are these required? They may not be if the customer wants to extract the user / pass with a custom function

Reply:
They are not required :)